packages/arb-ts/src/lib/message/L1ToL2Message.ts

```diff
@@ -237,54 +237,108 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     if (await this.isExpired()) {
       return L1ToL2MessageStatus.EXPIRED
     }
-    // we could sanity check that autoredeem failed, but we don't need to
-    // currently if the params (max l2 gas price) * (max l2 gas) = 0
-    // no auto-redeem receipt gets emitted at all; if the user cars about the difference
-    // between auto-redeem failed and auto-redeem never took place, they can check
-    // the receipt. But for the sake of this status method, NOT_YET_REDEEMED for both cases is okay.
-    return L1ToL2MessageStatus.NOT_YET_REDEEMED
+
+    // ticket was created but not redeemed
+    // this could be because
+    // a) the ticket is non auto redeemable (l2GasPrice == 0 || l2GasLimit == 0) -
+    //    this is usually an eth deposit. But in some rare case the
+    //    user may still want to manually redeem it
+    // b) the ticket is auto redeemable, but the auto redeem failed
+
+    // the fact that the auto redeem failed isn't usually useful to the user
+    // if they're doing an eth deposit they dont care about redemption
+    // and if they do want execution to occur they will know that they're
+    // here because the auto redeem failed. If they really want to check
+    // they can fetch the auto redeem receipt and check the status on it
+    return L1ToL2MessageStatus.FUNDS_DEPOSITED_ON_L2
   }
 
-  public async status(): Promise<L1ToL2MessageStatus> {
-    const l2TxReceipt = await this.getL2TxReceipt()
-    const retryableCreationReceipt = await this.getRetryableCreationReceipt()
+  protected async status(): Promise<L1ToL2MessageStatus> {
+    return this.receiptsToStatus(
+      await this.getRetryableCreationReceipt(),
+      await this.getL2TxReceipt()
+    )
+  }
 
-    return this.receiptsToStatus(retryableCreationReceipt, l2TxReceipt)
+  /**
+   * Wait for the retryable ticket to be created, and if it's an eth deposit also wait for the
+   * @param isOnlyEthDeposit Is the message only an eth deposit, and nothing else.
+   *  If the message deposits eth as well as calls a contract then false should be supplied,
+   *  or if this is just a contract call false should be supplied. True should be supplied if
+   *  if this message just deposits eth - e.g. by calling depositEth on the L1 inbox.
+   * @param timeout Amount of time to wait for the retryable ticket to be created
+   * @param confirmations Amount of confirmations the retryable ticket and the auto redeem receipt should have
+   * @returns The wait result contains a complete, a status, and optionally the l2TxReceipt.
+   * If complete is true then this message is in the terminal state for it's type. For eth deposits complete is
+   * true if the status is FUNDS_DEPOSITED, EXPIRED or REDEEMED. For all other messages complete is true
+   * only if the status is REDEEMED.
+   */
+  public async wait(
+    isOnlyEthDeposit: boolean,
+    timeout = 900000,
+    confirmations?: number
+  ): Promise<L1ToL2MessageWaitResult & { complete: boolean }> {
```

Comment:
```suggestion
  public async wait(
    isEthDeposit: boolean,
    confirmations?: number,
    timeout?: number
  ): Promise<L1ToL2MessageWaitResult & { complete: boolean }> {
```

A couple things here, i think we should handle timeouts the same way ethers does, [this](https://github.com/ethers-io/ethers.js/blob/73a46efea32c3f9a4833ed77896a216e3d3752a0/packages/providers/src.ts/base-provider.ts#L936) is the reference I found for it, that is to either have a user timeout or wait indefinitely.

Reply:
I'm also thinking it might be worth having `confirmations` as the first parameter since its usually the expected first param.
That is because someone consuming this in javascript might mis-use and mess it up

```suggestion
  public async wait(
    confirmations = 1,
    isEthDeposit = false,
    timeout = 0
  ): Promise<L1ToL2MessageWaitResult & { complete: boolean }> {
```
Maybe something like this would work

Reply:
The problem is I dont think we should have a default for `isEthDeposit`, since otherwise people wont be forced to think about it, and will get the wrong completion status.

Reply:
Re the timeout, lets have a chat about it. In this case we have a strong expectation of when we think a transaction will be included, so we want to be explain that to the user. That was the purpose of the default before, we could explain this in a comment?

Reply:
Re-ordered the params in `wait` and `waitForStatus`